File: app/src/main/java/org/helllabs/android/xmp/modarchive/SearchError.java
package org.helllabs.android.xmp.modarchive;

import java.util.Locale;

import org.helllabs.android.xmp.R;

import android.content.Intent;
import android.graphics.Typeface;
import android.os.Bundle;
import androidx.appcompat.app.AppCompatActivity;
import android.view.KeyEvent;
import android.widget.TextView;

public class SearchError extends AppCompatActivity implements Runnable {

	private static final int PERIOD = 1337;
	private static final String UNKNOWN_ERROR = "Software Failure.   Press back to continue.\n\nGuru Meditation #35068035.48454C50";

	private TextView msg;
	private boolean frameBlink;

	@Override
	public void onCreate(final Bundle savedInstanceState) {
		super.onCreate(savedInstanceState);

		//		// Hide the status bar
		//        if (Build.VERSION.SDK_INT < 16) {
		//            getWindow().setFlags(WindowManager.LayoutParams.FLAG_FULLSCREEN, WindowManager.LayoutParams.FLAG_FULLSCREEN);
		//        } else {
		//        	final View decorView = getWindow().getDecorView();
		//        	decorView.setSystemUiVisibility(View.SYSTEM_UI_FLAG_FULLSCREEN);
		//        }

		setContentView(R.layout.search_error);

		setTitle("Search error");

		final Throwable error = (Throwable)getIntent().getSerializableExtra(Search.ERROR);
		msg = (TextView)findViewById(R.id.error_message);
		//msg.getPaint().setAntiAlias(false);

		String message = error.getMessage();
		if (message == null) {
			message = UNKNOWN_ERROR;
		} else {
			// Remove java exception stuff
			final int idx = message.indexOf("Exception: ");
			if (idx >= 0) {
				message = message.substring(idx + 11);
			}

			if (message.trim().isEmpty()) {
				message = UNKNOWN_ERROR;
			} else {
				message = message.substring(0, 1).toUpperCase(Locale.US) + message.substring(1) + ".  Press back button to continue.";
			}
		}

		msg.setText(message);

		final Typeface typeface = Typeface.createFromAsset(getAssets(), "fonts/TopazPlus_a500_v1.0.ttf");
		msg.setTypeface(typeface);

		msg.postDelayed(this, PERIOD);
	}


	@Override
	public void onDestroy() {
		msg.removeCallbacks(this);
		super.onDestroy();
	}

	@Override
	public boolean onKeyDown(final int keyCode, final KeyEvent event) {

		// Back key returns to search
		if (keyCode == KeyEvent.KEYCODE_BACK) {
			final Intent intent = new Intent(this, Search.class);
			intent.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP);
			startActivity(intent);
			overridePendingTransition(0, 0);
			return true;
		}
		return super.onKeyDown(keyCode, event);
	}


	@SuppressWarnings("deprecation")
	@Override
	public void run() {
		// Guru frame blink
		msg.setBackgroundDrawable(getResources().getDrawable(frameBlink ? R.drawable.guru_frame : R.drawable.guru_frame_2));
		frameBlink ^= true;
		msg.postDelayed(this, PERIOD);		
	}
}

File: app/src/main/java/org/helllabs/android/xmp/modarchive/SearchError.kt
package org.helllabs.android.xmp.modarchive

import android.content.Intent
import android.os.Bundle
import android.view.KeyEvent
import androidx.appcompat.app.AppCompatActivity
import java.util.*
import org.helllabs.android.xmp.R
import org.helllabs.android.xmp.databinding.ActivitySearchErrorBinding
import org.helllabs.android.xmp.modarchive.ModArchiveConstants.ERROR

class SearchError : AppCompatActivity(), Runnable {

    private lateinit var binder: ActivitySearchErrorBinding

    private var frameBlink: Boolean = false

    public override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        binder = ActivitySearchErrorBinding.inflate(layoutInflater)

        setContentView(binder.root)
        setSupportActionBar(binder.appbar.toolbar)
        supportActionBar?.setDisplayHomeAsUpEnabled(true)
        supportActionBar?.setDisplayShowHomeEnabled(true)

        binder.appbar.toolbarText.text = getString(R.string.search_title_error)

        var message: String? = intent.getStringExtra(ERROR)
        if (message == null) {
            message = getString(R.string.search_unknown_error)
        } else {
            // Remove java exception stuff
            val idx = message.indexOf("Exception: ")
            if (idx >= 0) {
                message = message.substring(idx + 11)
            }
            message = if (message.trim { it <= ' ' }.isEmpty()) {
                getString(R.string.search_unknown_error)
            } else {
                val err = message.substring(0, 1).toUpperCase(Locale.US) + message.substring(1)
                getString(R.string.search_known_error, err)
            }
        }

        binder.errorMessage.text = message
    }

    override fun onResume() {
        super.onResume()
        binder.errorMessage.postDelayed(this, BLINK_PERIOD)
    }

    override fun onPause() {
        super.onPause()
        binder.errorMessage.removeCallbacks(this)
    }

    override fun onKeyDown(keyCode: Int, event: KeyEvent): Boolean {
        // Back key returns to search
        if (keyCode == KeyEvent.KEYCODE_BACK) {
            val intent = Intent(this, Search::class.java)
            intent.flags = Intent.FLAG_ACTIVITY_CLEAR_TOP
            overridePendingTransition(0, 0)
            startActivity(intent)
            return true
        }
        return super.onKeyDown(keyCode, event)
    }

    // Guru frame blink
    override fun run() {
        binder.errorMessage.apply {
            background.alpha = if (frameBlink) 255 else 0
            postDelayed(this@SearchError, BLINK_PERIOD)
        }

        frameBlink = !frameBlink
    }

    companion object {
        private const val BLINK_PERIOD = 1337L
    }
}
